analytics: Backend tests, fixture data, and other cleanups in preparation for 1.5.

rishig opened the pull request:
The commits through `analytics tests: Refactor assertCountEquals calls to be more readable.` only touch `test_counts.py`. After merging them we can also close PRs #3241 and #3280.

The commits through `analytics: Add fixtures for several CountStats.` add /stats views code, add fixtures, and do refactorings of `fixtures.py` and `populate_analytics_db.py`, but don't change anything fundamental.

The remaining 6 commits get rid of `BaseCount.interval`, and change frequencies of and merge various `COUNT_STATS`.

Don't have test coverage yet of `analytics/views.py`, but getting this out there so Amy can build on it (not planning on figuring out views tests before 1.5, unless you think otherwise).

smarx commented:
*Automated message from Dropbox CLA bot*

@rishig, it looks like you've already signed the Dropbox CLA. Thanks!

timabbott commented:
This seems pretty reasonable; there are a few potential concerns:
* I'm somewhat skeptical of https://github.com/zulip/zulip/pull/3307/commits/3845b44493b0e46c9b820550977c240f88e6fa0d from the standpoint that often the explicit (but verbose) thing can actually be more readable than a more finely encoded version.  I think it's probably worth adding a long-from block comment to that test class explaining how it works, at least.
* There's no tests for the new views, which seems fine for now but we'll want to address before too long.

This looks good; @rishig should I merge this now or did you have things you wanted to check before we merge this?

rishig commented:
Added the comment in 3845b44, and added one commit at the end to make the fixture data prettier.

Let's just merge this now, and expect views tests and bug fixes in upcoming commits. Adding @amyliu345 so she gets notified when this is merged.

timabbott commented:
Merged, thanks @rishig!
